src/SmartComponents/DriftPage/DriftTable/ComparisonHeader/ComparisonHeader.js

```diff
@@ -12,33 +12,44 @@ import { ASC, DESC } from '../../../../constants';
 import HistoricalProfilesPopover from '../../../HistoricalProfilesPopover/HistoricalProfilesPopover';
 import ReferenceSelector from '../ReferenceSelector/ReferenceSelector';
 
-class ComparisonHeader extends Component {
-    constructor(props) {
-        super(props);
-        this.columnWidth = React.createRef();
-
-        this.state = {
-            refState: null
-        };
-    }
-
-    setColumnWidth = () => {
-        if (this.columnWidth?.current) {
-            this.props.setColumnHeaderWidth(this.columnWidth.current.offsetWidth);
-            this.setState({ refState: this.columnWidth });
+const ComparisonHeader = ({
+    factSort,
+    fetchCompare,
+    masterList,
+    permissions,
+    referenceId,
+    removeSystem,
+    selectedBaselineIds,
+    selectedHSPIds,
+    selectHistoricProfiles,
+    setColumnHeaderWidth,
+    setHistory,
+    stateSort,
+    systemIds,
+    toggleFactSort,
+    toggleStateSort,
+    updateReferenceId
+}) => {
+    const columnWidth = useRef();
+    const [ refState, setRefState ] = useState(null);
+
+    const setColumnWidth = () => {
+        if (columnWidth?.current) {
+            setColumnHeaderWidth(columnWidth.current.offsetWidth);
+            setRefState({ refState: columnWidth });
         }
     };
 
-    componentDidMount() {
-        this.setColumnWidth();
-        window.addEventListener('resize', debounce(this.setColumnWidth, 250));
-    }
+    useEffect(() => {
+        setColumnWidth();
+        window.addEventListener('resize', debounce(setColumnWidth, 250));
```

Comment:
It would maybe be good to add a cleanup function here. 

https://www.pluralsight.com/guides/how-to-cleanup-event-listeners-react#module-manualeventlistenercleanup